sys/boot/geli/geliboot.c

```diff
@@ -217,20 +268,22 @@ geli_attach(struct dsk *dskp, const char *passphrase)
 		error = g_eli_mkey_decrypt(&geli_e->md, key, mkey, &keynum);
 		if (error == -1) {
 			bzero(&mkey, sizeof(mkey));
-                        bzero(&key, sizeof(key));
-			printf("Bad GELI key: %d\n", error);
+			bzero(&key, sizeof(key));
+			printf("Bad GELI key: bad password?\n");
```

Comment:
the printf() in boot2 cannot do signed integers. Also, this matches the error message from GELI itself